src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,206 @@
+        public Channel ChargeChannel
+        {
+            get => _chargeChannel;
+            set
+            {
+                if ((int)value >= 0 & (int)value < 14)
+                {
+                    _chargeChannel = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address pin</param>
+        /// <param name="chipSelect">Chip Select pin</param>
+        /// <param name="dataOut">Data Out pin</param>
+        /// <param name="inputOutputClock">I/O Clock pin</param>
+        /// <param name="endOfConversion">End of Conversion pin</param>
+        /// <param name="controller">The GPIO controller for defined external pins. If not specified, the default controller will be used.</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion = -1, GpioController controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = controller == null || shouldDispose;
+            _controller = controller ?? new GpioController(pinNumberingScheme);
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _endOfConversion = endOfConversion;
+            _inputOutputClock = inputOutputClock;
+
+            _controller.OpenPin(_address, PinMode.Output);
+            _controller.OpenPin(_chipSelect, PinMode.Output);
+            _controller.OpenPin(_dataOut, PinMode.InputPullUp);
+            _controller.OpenPin(_inputOutputClock, PinMode.Output);
+            if (_endOfConversion != -1)
+            {
+                _controller.OpenPin(_endOfConversion, PinMode.InputPullUp);
+            }
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Reads sensor value in Fast Mode 1 (10 clock transfer using !chipSelect)
+        /// <remarks>
+        /// <br>1 cycle used - gets data on second call of this method</br>
+        /// <br>First cycle: Ask for value on the channel</br>
+        /// </remarks>
+        /// </summary>
+        /// <param name="channelNumber">Channel to read</param>
+        /// <returns>10 bit value corresponding to relative voltage level on channel</returns>
+        public int ReadChannel(int channelNumber)
+        {
+            if (channelNumber < 0 & channelNumber > 14)

```

Comment:
you should create helper method like `IsValidChannelNumber`. Here you include 14 as valid while previously you checked for:
`if ((int)value >= 0 & (int)value < 14)` which made 14 invalid

so you should pick which one is true - in order to prevent this kind of errors it's better to have a method for that